Apps/Admin/Common/Models/AdminAgent.cs

```diff
@@ -41,8 +40,8 @@ public class AdminAgent
         public KeycloakIdentityProvider IdentityProvider { get; set; } = KeycloakIdentityProvider.Unknown;
 
         /// <summary>
-        /// Gets or sets the roles assigned to the agent.
+        /// Gets the roles assigned to the agent.
         /// </summary>
-        public IEnumerable<IdentityAccessRole> Roles { get; set; } = Enumerable.Empty<IdentityAccessRole>();
+        public ISet<IdentityAccessRole> Roles { get; init; } = new HashSet<IdentityAccessRole>();
```

Comment:
Is switching IEnumerable to ISet required?

Reply:
It enabled easier checking of one of the requirements on the front-end since it has no notion of order and it seems to better reflect the nature of the data.